lib/game.rb

```diff
@@ -0,0 +1,72 @@
+require_relative 'frame'
+
+class Game
+  attr_reader :frames
+  
+  def initialize
+    @frames = [Frame.new]
+  end
+
+  def roll(pins)
+    case 
+    when frame_count == 1 && current_frame.in_play
+         current_frame.knocked(pins)
+    when frame_count <= 8 && (current_frame.in_play == false)
+         new_frame
+         current_frame.knocked(pins)
+         point_bonus(pins) if (last_frame_strike?) && (pins == 10)
+         point_bonus(pins) if last_frame_strike?
+         point_bonus(pins) if last_frame_spare?
+    when frame_count <=8 && current_frame.in_play
+         current_frame.knocked(pins)
+         point_bonus(pins) if last_frame_strike?
+    when frame_count == 9 && (current_frame.in_play == false)
+         new_frame_10
+         current_frame.knocked(pins)
+         point_bonus(pins) if (last_frame_strike?) && (pins == 10)
+         point_bonus(pins) if last_frame_strike?
+         point_bonus(pins) if last_frame_spare?
+    else frame_count == 10 && current_frame.in_play
+         current_frame.knocked(pins)
+    end
+  end
+
+  def total_score
+    total_score = @frames.map {|frame | frame.score }.sum
+  end
+
+  def last_frame_strike?
```

Comment:
It seems like this method could be made private because we're not using it outside of this class and it's not a method that the end-user would need to call themselves. Are there other methods in both this and the Frame class which could be made private?